omod-common/src/main/java/org/openmrs/module/webservices/rest/web/resource/impl/BaseDelegatingResource.java

```diff
@@ -263,6 +263,27 @@ public T newInstance(String type) {
 	 */
 	protected abstract void delete(T delegate, String reason, RequestContext context) throws ResponseException;
 	
+	/**
+	 * Unvoid or unretire delegate, whichever action is appropriate for the resource type.
+	 * Subclasses need to override this method, which is called internally by
+	 * {@link #unDelete(String, RequestContext)}.
+	 * 
+	 * @param delegate
+	 * @param context
+	 * @throws ResponseException
+	 * @return Object
+	 */
+	
+	// TODO make this method an abstruct method when all subclasses have overriden it
```

Comment:
Can't merge code that has a TODO in it.  Remove this comment (and the blank lines).

If there is no base implementation of this the method should be abstract, just like delete. Are there many other resources that would need this fixed?

Reply:
I deleted the TODO and empty line but left the default implementation in place because I think this will help the end user trouble shoot problems that may come with existing resources that may not be part of the rest.webservices module and don't have their own implementation for this already.